src/Xamarin.Android.Build.Utilities/AndroidSdk.cs

```diff
@@ -100,11 +100,7 @@ public static string GetPlatformDirectoryFromApiLevel (string apiLevel, AndroidV
 				return dir;
 
 			var level = versions.GetApiLevelFromId (id);
-			dir       = level.HasValue ? GetPlatformDirectory (level.Value) : null;
-			if (dir != null && Directory.Exists (dir))
-				return dir;
-
-			return null;
+			return level.HasValue ? GetPlatformDirectory (level.Value) : null;
```

Comment:
I thought our discussed solution was to *not* return `null` -- at all -- as that was responsible for the `ArgumentNullException`.

If `level.HasLevel` is `false`, then we should return `dir`:

```csharp
var level = versions.GetApiLevelFromId (id);
if (level == null)
    return dir;
return GetPlatformDirectory (level.Value);
```